skimage/segmentation/_join.py

```diff
@@ -3,7 +3,7 @@
 from ..util._map_array import map_array, ArrayMap
 
 
-def join_segmentations(s1, s2):
+def join_segmentations(s1, s2, return_mapping: bool = False):
```

Comment:
What do you guys think about having two functions to avoid the "may return different output lengths"-mess? E.g.

```python
def join_segmentation_map(s1, s2):
    ...
    return j, map_j_to_s1, map_j_to_s2

def join_segmentations(s1, s2):
    j, *_ = join_segmentation_map(s1, s2)
    return j
```

Regardless of how we decide to tackle this issue in the future, I think this approach is easier to update with deprecations and doesn't introduce a lot of duplication, besides the docstrings.

Reply:
Nope, hate it LOL. 

Since the default is False I don't think deprecations text backwards compatibility are an issue. Going forward, I think either approach is equally likely to result in a deprecation. And the two functions are (imho) ugly and I don't see us keeping it long term. One close alternative would be join_segmentations vs extra.join_segmentations (ie new namespace, same function name), but then you've still got a deprecation anyway.

Either way, I think this is a much bigger discussion that shouldn't hold up this PR. For now, `return_foo=` is actually the most consistent with the rest of the library, so I think we should stick with it.

Reply:
I can live with that. :smile: I find it interesting that you have such a strong reaction to it. For me that pattern seems way more expressive and easier to reason about / test. But I agree that's an interesting discussion for another time.